src/mds/Server.cc

```diff
@@ -7002,6 +7002,17 @@ void Server::handle_client_mknod(const MDRequestRef& mdr)
   else
     layout = mdcache->default_file_layout;
 
+  if (!layout.is_valid()) {
```

Comment:
Seems reasonable as the same check is performed in handle_client_openc(). Quick question - did you run into an issues due to the lack of this check? @jhonxue

Reply:
Yes，encountered some situations, but not sure they are related. 

standby-reply mds:
           2024-01-11T07:56:13.731+0000 7f0077ca3700  0 mds.0.journal EMetaBlob.replay invalid layout on ino [inode 0x4003257a7c4 [2,head] #4003257a7c4 auth v1851322605 s=0 nl=0 n(v1079 rc2024-01-11T07:56:12.815778+0000 b67737072290 6452=6447+5) (iversion lock) 0x5610be080c00]: 000-1""
2024-01-11T07:56:13.732+0000 7f0077ca3700 -1 log_channel(cluster) log [ERR] : Invalid layout for inode 0x4003257a7c4 in journal
2024-01-11T07:56:14.640+0000 7f0077ca3700  1 mds.a.yj-bstoreceph01rack07meta-01.cmqeqz respawn!

Standby-reply mds report to  mon, mon blocklisted the active mds.

Active-mds :
             2024-01-11T07:56:13.988+0000 7f1f6fc2d700 -1 mds.0.journaler.mdlog(rw) _finish_flush got (108) Cannot send after transport endpoint shutdown
2024-01-11T07:56:13.988+0000 7f1f6fc2d700 -1 mds.0.journaler.mdlog(rw) handle_write_error (108) Cannot send after transport endpoint shutdown
2024-01-11T07:56:13.988+0000 7f1f6fc2d700 -1 MDSIOContextBase: failed with -108, restarting...
2024-01-11T07:56:13.988+0000 7f1f6fc2d700  1 mds.a.yj-bstoreceph01rack08meta-01.jgtzrd respawn!

Reply:
If the standby-replay mds reports damaged to the mon, then the mon can blocklist the active mds. Did that happen?

Reply:
Active-mds log:  active-mds can not flush journal to the rados

2024-01-11T07:56:13.988+0000 7f1f6fc2d700 -1 mds.0.journaler.mdlog(rw) _finish_flush got (108) Cannot send after transport endpoint shutdown
2024-01-11T07:56:13.988+0000 7f1f6fc2d700 -1 mds.0.journaler.mdlog(rw) handle_write_error (108) Cannot send after transport endpoint shutdown
2024-01-11T07:56:13.988+0000 7f1f6fc2d700 -1 MDSIOContextBase: failed with -108, restarting...
2024-01-11T07:56:13.988+0000 7f1f6fc2d700 1 mds.a.yj-bstoreceph01rack08meta-01.jgtzrd respawn!

Reply:
mon log:  blocklisted mds

             2024-01-11T07:56:13.733+0000 7f89cb99e700  5 mon.yj-bstoreceph01rack05meta-01@0(leader).mds e286311 prepare_beacon mds.0 up:standby-replay -> down:damaged

2024-01-11T07:56:13.733+0000 7f89cb99e700  0 mon.yj-bstoreceph01rack05meta-01@0(leader).mds e286311 prepare_beacon: marking rank 0 damaged

2024-01-11T07:56:13.733+0000 7f89cb99e700  1 mon.yj-bstoreceph01rack05meta-01@0(leader).mds e286311 fail_mds_gid 1359886760 mds.a.yj-bstoreceph01rack07meta-01.cmqeqz role 0



bool MDSMonitor::fail_mds_gid(FSMap &fsmap, mds_gid_t gid)
{
  const auto& info = fsmap.get_info_gid(gid);
  dout(1) << "fail_mds_gid " << gid << " mds." << info.name << " role " << info.rank << dendl;

  ceph_assert(mon.osdmon()->is_writeable());

  epoch_t blocklist_epoch = 0;
  if (info.rank >= 0 && info.state != MDSMap::STATE_STANDBY_REPLAY) {
    utime_t until = ceph_clock_now();
    until += g_conf().get_val<double>("mon_mds_blocklist_interval");
    blocklist_epoch = mon.osdmon()->blocklist(info.addrs, until);
  }

  fsmap.erase(gid, blocklist_epoch);
  last_beacon.erase(gid);
  if (pending_daemon_health.count(gid)) {
    pending_daemon_health.erase(gid);
    pending_daemon_health_rm.insert(gid);
  }

  return blocklist_epoch != 0;
}